### PR TITLE
Use 'groupAccess' instead of chmod for changing SCC permissions

### PR DESCRIPTION
**Pull request by ashu-mehra**
Using chmod in populate_scc.sh to change the permission causes all
the files in the directory in the layers of the docker images
to be duplicated in the current docker layer. This prevents sharing of the 
common SCC layer between different application docker images when 
they use this script to create application specific scc layer.
This change removes the use of chmod and uses 'groupAccess' sub-option
of -Xshareclasses to provide group read-write access to the cache.
However, this option requires umask to be set up properly (to 002) to be
able to set group read-write access on the SCC file.
More information about groupAccess sub-option can be seen at:
https://www.eclipse.org/openj9/docs/xshareclasses/#groupaccess

Signed-off-by: Ashutosh Mehra <mehra.ashutosh@ibm.com>

### CODE REVIEW

**Comment by ashu-mehra** [comment]
@arthurdm can you please review.

fyi - @ymanton @vijaysun-omr

**Comment by ashu-mehra** [comment]
@ymanton can you please review these changes.

**Comment by ashu-mehra** [comment]
I have no clue either. But I noticed following comment in dockerfile [1], not sure if that is the reason for granting group rw access to directories.

```
#These settings are needed so that we can run as a different user than 1001 after server warmup
```

[1] https://github.com/OpenLiberty/ci.docker/blob/master/releases/latest/kernel/Dockerfile.ubi.ibmjava8#L81

**Comment by arthurdm** [comment]
hey guys - ya, the group permissions is to enable these images to run on OpenShift using their default policy of a random uid + gid of 0.

**Comment by arthurdm** [comment]
hi @ashu-mehra - can you please confirm if you have tested this change on OpenShift?

File: releases/latest/kernel/helpers/build/populate_scc.sh [hunk 1]
```diff
@@ -89,7 +92,10 @@ do
   /opt/ol/wlp/bin/server start && /opt/ol/wlp/bin/server stop
 done
 
-rm -rf /output/messaging /logs/* $WLP_OUTPUT_DIR/.classCache && chmod -R g+rwx /opt/ol/wlp/output/*
```
Comment (arthurdm) [inline releases/latest/kernel/helpers/build/populate_scc.sh:92]: are you sure we don't need `chmod -R g+rwx /opt/ol/wlp/output/*` anymore?   CC @arturdzm

**Comment by ashu-mehra** [comment]
> can you please confirm if you have tested this change on OpenShift?

@arthurdm When I tested these changes on OpenShift the java process inside the container is running with uid 1001. As per your [comment](https://github.com/OpenLiberty/ci.docker/pull/166#issuecomment-619235703) shouldn't it be running with some random uid?

File: releases/latest/kernel/helpers/build/populate_scc.sh (hunk 1 above)
Comment (arturdzm) [reply releases/latest/kernel/helpers/build/populate_scc.sh:92]: Assuming that all files created by the server will have `rw` permissions not just `r` I think its ok. There could be files copied by the user into that folder though that are not readable/writable by the group. Eg would be keystore p12 file. That depends on what permissions the files had on a host

**Comment by arturdzm** [comment]
@ashu-mehra It might be running with `anyuid` scc in your case. you can check it on pod annotations I think.

**Comment by ashu-mehra** [comment]
@arturdzm this is the output I get using `oc describe project <name>`:
```
Annotations:            openshift.io/description=
                        openshift.io/display-name=
                        openshift.io/requester=kube:admin
                        openshift.io/sa.scc.mcs=s0:c24,c4
                        openshift.io/sa.scc.supplemental-groups=1000560000/10000
                        openshift.io/sa.scc.uid-range=1000560000/10000
```
I don't see `anyuid` in the output.

Nevertheless, assuming it is running with `anyuid`, what can I do to make the containers use random uid to check `groupAccess` option is working as expected.

**Comment by arturdzm** [comment]
The annotation would be on the pods not project.
It depends on ServiceAccount and what SCC it was granted.
You might have to create new SA or remove scc using `oc adm policy...`
As the project does say `uid-range=1000560000`. I am not sure if you're deploying using operator or manually (operator would create new SA for every app).

**Comment by ashu-mehra** [comment]
Used `oc adm policy remove-scc-from-group anyuid system:authenticated` to remove `anyuid` scc. With this change, I see the java processes are running with a different uid (`1000590000`), and I can still see the common layer of shared class cache is being shared between the two pods running different application.
I guess that confirms `groupAccess` is working and the sharing is taking place as expected.

**Pull request merged**